docs/source/cli_help/clean.rst

```diff
@@ -0,0 +1,24 @@
+.. _clean:
+
+clean
+----
+**Usage:**
+
+.. code-block:: shell
+
+    git machete clean [-c|--checkout-my-github-prs]
+
+If invoked without any flag, deletes untracked and unmanaged branches.
+
+If invoked with ``-c`` or ``--checkout-my-github-prs``, checks out your open PRs into local branches.
+
```

Comment:
I'd mention it's equivalency to `github sync`, and vice versa, mention `clean` from help of `github sync`